crowbar_framework/app/models/pacemaker_service.rb

```diff
@@ -369,6 +369,28 @@ def apply_role_pre_chef_call(old_role, role, all_nodes)
     role.default_attributes["corosync"]["members"] = member_nodes.map{ |n| n.get_network_by_type("admin")["address"] }
     role.default_attributes["corosync"]["transport"] = role.default_attributes["pacemaker"]["corosync"]["transport"]
 
+    # Set up the second ring if dual_ring mode configured
+    ring_mode = role.default_attributes["pacemaker"]["corosync"]["ring_mode"]
+
+    if ring_mode == "dual_ring"
+      second_ring_network = role.default_attributes["pacemaker"]["corosync"]["second_ring_network"]
+      second_ring_net = Chef::DataBag.load("crowbar/#{second_ring_network}_network")
+
+      net_svc = NetworkService.new @logger
+      second_ring_members = member_nodes.map do |member_node|
+        allocated_ip_response = net_svc.allocate_ip "default", second_ring_network, "host", member_node.name
+        allocated_ip_response[1]["address"]
+      end
+
+      role.default_attributes["corosync"]["second_ring_used"] = true
+      role.default_attributes["corosync"]["second_ring_bind_addr"] = second_ring_net["network"]["subnet"]
+      role.default_attributes["corosync"]["second_ring_mcast_addr"] = role.default_attributes["pacemaker"]["corosync"]["mcast_addr"]
+      role.default_attributes["corosync"]["second_ring_mcast_port"] = role.default_attributes["pacemaker"]["corosync"]["mcast_port"]
+      role.default_attributes["corosync"]["second_ring_members"] = second_ring_members
```

Comment:
Is it just me or do you now have both `[:corosync][:second_ring_used]` and `[:pacemaker][:corosync][:second_ring_used]`?  And I don't see why either are needed - they seem to be duplicating the purpose of `ring_mode`.